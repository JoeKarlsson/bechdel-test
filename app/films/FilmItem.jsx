import React from 'react';
import { Link } from 'react-router';
import styles from './FilmItem.scss';

class FilmItem extends React.Component {
  render() {
    return (
      <div className='filmItem'>
        <li>
          <div>
            <Link
              className='gist_description'
              to={`/film/${this.props.film._id}`}
            >
              <img
                src={this.props.film.images.poster}
                alt={this.props.film.title}
              />
              <p>
                {this.props.film.title}
              </p>
            </Link>
          </div>
        </li>
      </div>
    );
  }
}

FilmItem.propTypes = {
  film: React.PropTypes.object,
};

FilmItem.defaultProps = {
  film: {
    title: 'Movie Title',
    images: {
      poster: 'Movie Poster URL',
    },
    id: '123456',
  },
};

export default FilmItem;
