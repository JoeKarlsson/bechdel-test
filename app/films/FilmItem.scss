$tomato: tomato;

.filmItem {
  background-color: $tomato;
}

.gist_description {
  background-color: green;
}