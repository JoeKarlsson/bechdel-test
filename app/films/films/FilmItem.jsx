import React from 'react';
import { Link } from 'react-router';
import styles from './FilmItem.scss';

class FilmItem extends React.Component {
  render() {
    return (
      <div className={styles.filmItem}>
        <li className={styles.filmItem_ListItem}>
          <div>
            <Link
              className={styles.gist_description}
              to={`/film/${this.props.film._id}`}
            >
              <img
                src={this.props.film.images.poster}
                alt={this.props.film.title}
              />
              <p className={styles.filmItem_title}>
                {this.props.film.title}
              </p>
            </Link>
          </div>
        </li>
      </div>
    );
  }
}

FilmItem.propTypes = {
  film: React.PropTypes.object,
};

FilmItem.defaultProps = {
  film: {
    title: 'Movie Title',
    images: {
      poster: 'Movie Poster URL',
    },
    id: '123456',
  },
};

export default FilmItem;
