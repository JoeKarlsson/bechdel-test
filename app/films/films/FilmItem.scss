.filmItem {

}

.gist_description {

}

.filmItem_ListItem {
  list-style-type: none;
}

.filmItem_title {
  text-align: center;
  color: black;
}