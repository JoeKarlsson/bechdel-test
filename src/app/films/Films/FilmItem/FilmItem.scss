.filmItem {
	background-image: url(http://placehold.it/170x240);
	background-size: 100% 100%;
	display: inline-block;
	height: 240px;
	width: 170px;
	font-size: 10pt;
	margin: 2.5px;
	color: #000;
	position: relative;
	cursor: pointer;

	.description {
		position: absolute;
		bottom: 0;
		left: 0;
		right: 0;
		height: 0px;
		padding: 0px;
		background-color: rgba(0, 0, 0, 0.85);
		color: white;
		overflow: hidden;
		transition: all 0.3s;
	}

	&:hover {
		height: 30px;
		padding: 10px;
	}

	h2 {
		color: #33cc33;
		font-weight: bold;
	}

	li {
		list-style-type: none;
	}

	.filmItem_title {
		text-align: center;
		color: black;
	}
}
