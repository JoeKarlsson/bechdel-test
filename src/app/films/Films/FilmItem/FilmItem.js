import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './FilmItem.scss';

const FilmItem = props => {
	const { film } = props;
	const filmUrl = `/film/${film._id}`;

	return (
		<div className="filmItem">
			<li>
				<div>
					<Link to={filmUrl}>
						<img src={film.images.poster} alt={film.title} />
						<div className="description">
							<h2 className="filmItem_title">{film.title}</h2>
							<p>2015</p>
						</div>
					</Link>
				</div>
			</li>
		</div>
	);
};

FilmItem.propTypes = {
	film: PropTypes.object, // eslint-disable-line react/forbid-prop-types
};

FilmItem.defaultProps = {
	film: {
		title: 'Movie Title',
		images: {
			poster: 'Movie Poster URL',
		},
		id: '123456',
	},
};

export default FilmItem;
