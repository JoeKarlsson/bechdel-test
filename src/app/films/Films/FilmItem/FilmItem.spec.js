import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilmItem from './FilmItem';

configure({ adapter: new Adapter() });

describe('FilmItem', () => {
	let wrapper;

	global.requestAnimationFrame = callback => {
		setTimeout(callback, 0);
	};

	beforeEach(() => {
		wrapper = shallow(<FilmItem />);
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});
		});
	});
});
