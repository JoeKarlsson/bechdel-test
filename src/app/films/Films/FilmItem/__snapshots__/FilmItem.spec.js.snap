// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmItem rendering initial state is rendered 1`] = `
<div
  className="filmItem"
>
  <li>
    <div>
      <a
        href="/film/undefined"
        onClick={[Function]}
      >
        <img
          alt="Movie Title"
          src="Movie Poster URL"
        />
        <div
          className="description"
        >
          <h2
            className="filmItem_title"
          >
            Movie Title
          </h2>
          <p>
            2015
          </p>
        </div>
      </a>
    </div>
  </li>
</div>
`;
